src/TouchCarousel.js

```diff
@@ -55,22 +55,37 @@ class TouchCarousel extends React.PureComponent {
     this.grabbing = false
     this.tracingTouchId = null
     this.isMovingCross = null
+    this.isMouseDown = false
+    this.mouseTouchId = 0
   }
 
   componentDidMount () {
+    document.addEventListener('mousemove', this.onMouseMove)
+    document.addEventListener('mouseup', this.onMouseUp)
     this.autoplayIfEnabled()
   }
 
   componentWillUnmount () {
+    document.removeEventListener('mousemove', this.onMouseMove)
+    document.removeEventListener('mouseup', this.onMouseUp)
     this.stopAutoplay()
   }
 
+  onMouseDown = (e) => {
+    this.isMouseDown = true
+    this.onStart(1, ++this.mouseTouchId, e)
+  }
+
   onTouchStart = (e) => {
+    this.onStart(e.changedTouches.length, getTouchId(e), e)
+  }
+
+  onStart = (changeCount, touchId, e) => {
```

Comment:
What about keeping the original `onTouchStart`, `onTouchMove`, `onTouchEndOrCancel` untouched and mock up some events in the mouseEvent handlers and pass the mocked events to the touchEvent handlers? Maybe it's more clear and we keep touch support as first class.

Reply:
To be honest I did it this way because I didn't want to mock anything - I feel that explicit parameters are clearer than mocking touch events. If you look at https://github.com/xiaody/react-touch-carousel/pull/4/commits/6f5d76b42e6bf5004ca84fcd343835a7847166bd, you'll see that the changes are quite minimal, `onStart` is practically the same as the old `onTouchStart`, with 2 lines of code moved to the new `onTouchStart`

Reply:
Yes, I see the changes are minimal.
But for me, the `onStart(changeCount, touchId, e)` signature looks quite confusing:

- changeCount: why does it need a count?
- touchId: umm ok this makes sense
- e: oh finally you need still the `e` so why don't you just calculate the count and id by yourself?

And when look into the function body, it's still all about "touch".
If you stick to this way, you'd better rename all the things from "touch" to say "pointer".

Reply:
I'd prefer mocking touch events anyway.